features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/source/src/app/components/Apis/Listing/ApiTableRow.js

```diff
@@ -16,63 +16,223 @@
+    }));
+  };
 
-    handleApiDelete(e) {
-        this.setState({loading: true});
-        const api = new API();
-        const api_uuid = this.props.api.id;
-        const name = this.props.api.name;
-        let promised_delete = api.deleteAPI(api_uuid);
-        promised_delete.then(
-            response => {
-                if (response.status !== 200) {
-                    console.log(response);
-                    message.error("Something went wrong while deleting the " + name + " API!");
-                    this.setState({open: false});
-                    return;
-                }
-                message.success(name + " API deleted successfully!");
-                this.setState({active: false, loading: false});
-            }
+  handleRequestClose = () => {
+    this.setState({ openUserMenu: false });
+  };
+
+  handleApiDelete(e) {
+    this.setState({ loading: true });
+    const api = new API();
+    const api_uuid = this.props.api.id;
+    const name = this.props.api.name;
+    let promised_delete = api.deleteAPI(api_uuid);
+    promised_delete.then(response => {
+      if (response.status !== 200) {
+        console.log(response);
+        message.error(
+          "Something went wrong while deleting the " + name + " API!"
         );
-    }
+        this.setState({ open: false });
+        return;
+      }
+      message.success(name + " API deleted successfully!");
+      this.setState({ active: false, loading: false });
+    });
+  }
 
-    render() {
-        let details_link = "/apis/" + this.props.api.id;
-        const {name, version, context, description} = this.props.api;
-        if (!this.state.active) { // Controls the delete state, We set the state to inactive on delete success call
-            return null;
-        }
-        return (
-            <TableRow hover>
-                <TableCell><Link to={details_link}>{name}</Link></TableCell>
-                <TableCell>{version}</TableCell>
-                <TableCell>{context}</TableCell>
-                <TableCell style={{ whiteSpace: "normal", wordWrap: "break-word"}}>
-                    <Typography>
-                        {description}
-                    </Typography>
-                </TableCell>
-            </TableRow>
+  componentDidMount() {
+    const api = new API();
+    // Get api details, subscribed applications and user rating
+    let promised_api = api.getAPIById(this.props.api.id);
+    let existing_subscriptions = api.getSubscriptions(this.props.api.id, null);
+    let promised_applications = api.getAllApplications();
+    let promised_rating = api.getRatingFromUser(this.props.api.id, null);
+
+    Promise.all([
+      promised_api,
+      existing_subscriptions,
+      promised_applications,
+      promised_rating
+    ])
+      .then(response => {
+        let [api, subscriptions, applications, rating] = response.map(
+          data => data.obj
+        );
+        this.setState({ policies: api.policies });
```

Comment:
I think we could simplify the 100, 101 and 102 lines as follows

```javascript
this.setState({ api.policies, api.description, rating: rating.userRating });
```

Reply:
Fixed.